### PR TITLE
Domain event

### PR DESCRIPTION
**Pull request by EVDW**

### CODE REVIEW

File: _includes/_components/domain-event.html [hunk 1]
```diff
@@ -1,50 +1,133 @@
...
             <h2>Domain Event</h2>
-            <p>This component blablabla</p>
-          </div>
-        </header>
-        <div class="container content">
-          <p>
-            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
-          </p>
-          <div class="row">
+            <p>The raise method allows you to trigger any event on your entity and dispatched this Knp\Rad\DomainEvent\Event once the entity has been flushed.</p>
```
Comment (einenlum) [inline _includes/_components/domain-event.html]: `The raise method allows you to trigger any event in your entity and a Knp\Rad\DomainEvent\Event once the entity is flushed.`

File: _includes/_components/domain-event.html [hunk 2]
```diff
@@ -1,50 +1,133 @@
...
+                <div class="standard">
+                    <div class="banner"></div>
+{% highlight php %}
+<?php
+
+namespace App\Entity;
+
+class MyEntity
+{
+    // ...
```
Comment (einenlum) [inline _includes/_components/domain-event.html]: add `private $param;`

File: _includes/_components/domain-event.html [hunk 3]
```diff
@@ -1,50 +1,133 @@
...
+
+    public function setParam($value)
+    {
+        // ...
```
Comment (einenlum) [inline _includes/_components/domain-event.html]: `$this->param = $param;`

File: _includes/_components/domain-event.html [hunk 4]
```diff
@@ -1,50 +1,133 @@
...
+
+class MyEventListener
 {
-    public function test(SomeClass $subject)
+    public function preUpdate(PreUpdateEventArgs $event)
     {
-        return $subject->doSomething();
+        $entity = $event->getEntity();
+
+        if (!$entity instanceof MyEtity &&
```
Comment (einenlum) [inline _includes/_components/domain-event.html]: `MyEntity`

Comment (einenlum) [inline _includes/_components/domain-event.html]:
``` php
if (!$entity instanceof MyEntity) {
    return;
}

if ($event->hasChangedField('param')) {
    // do some stuff
}
```

File: _includes/_components/domain-event.html [hunk 5]
```diff
@@ -1,50 +1,133 @@
...
+class MyEntity implements DomainEvent\Provider
+{
+    // ...
+
+    public function setParam($arg)
+    {
+        // ...
+
+        $this->raise('myEventName', [
+            'anyKey' => $anyValue
```
Comment (einenlum) [inline _includes/_components/domain-event.html]: `'anyValue'`

File: _includes/_components/domain-event.html [hunk 6]
```diff
@@ -1,50 +1,133 @@
...
 <?php
 
-class Test
+namespace App\Entity;
+
+use Knp\Rad\DomainEvent;
+
+class MyEntity implements DomainEvent\Provider
+{
+    // ...
```
Comment (einenlum) [inline _includes/_components/domain-event.html]: The entity has also to use a specific trait.

`use DomainEvent\ProviderTrait;`

File: _includes/_components/domain-event.html (hunk 5 above)
Comment (EVDW) [inline _includes/_components/domain-event.html]: Hein?

**Comment by einenlum** [comment]
Please squash.

**Pull request merged**